src/server.ts

```diff
@@ -27,19 +49,20 @@ app.get('/api/credentials', async (_request, response) => {
   }
 });
 
-app.post('/api/credentials', async (request, response) => {
-  const credential: Credential = request.body;
-  await addCredential(credential);
-  response.status(200).send(credential);
-});
-
 //.put updates whole cred, .patch only parts
 app.put('/api/credentials/:service', async (request, response) => {
-  // the {} is destructuring
   const { service } = request.params;
   const credential: Credential = request.body;
+  const masterPassword = request.headers.authorization;
+  if (!masterPassword) {
+    response.status(400).send('Nope, kindly f*ck off');
+    return;
+  } else if (!(await validateMasterpassword(masterPassword))) {
+    response.status(401).send('You shall not pass');
```

Comment:
Not very informative API responses

Reply:
Sorry, it's been late last night and I was frustrated by this until it clicked. I planned to change it but forgot 😅